cluster/manifests/skipper/deployment.yaml

```diff
@@ -193,122 +219,127 @@ spec:
-          - "-oauth2-token-cookie-name={{ .ConfigItems.skipper_oauth2_cookie_name }}"
+          - "-oauth2-token-cookie-name={{ .Cluster.ConfigItems.skipper_oauth2_cookie_name }}"
           - "-oauth2-token-cookie-remove-subdomains=0"
-          - "-oauth2-callback-path={{ .ConfigItems.skipper_oauth2_redirect_uri_path }}"
-          - "-oauth2-grant-tokeninfo-keys={{ .ConfigItems.skipper_oauth2_ui_login_tokeninfo_keys }}"
+          - "-oauth2-callback-path={{ .Cluster.ConfigItems.skipper_oauth2_redirect_uri_path }}"
+          - "-oauth2-grant-tokeninfo-keys={{ .Cluster.ConfigItems.skipper_oauth2_ui_login_tokeninfo_keys }}"
 {{ end }}
-{{ if eq .ConfigItems.skipper_open_policy_agent_enabled "true" }}
+{{ if eq .Cluster.ConfigItems.skipper_open_policy_agent_enabled "true" }}
           - "-enable-open-policy-agent"
           - "-open-policy-agent-config-template=/etc/skipper/open-policy-agent/opaconfig.yaml"
           - "-open-policy-agent-envoy-metadata=/etc/skipper/open-policy-agent/envoymetadata.json"
 {{ end }}
-{{ if or (eq .ConfigItems.nlb_switch "pre") (eq .ConfigItems.nlb_switch "exec") }}
+{{ if or (eq .Cluster.ConfigItems.nlb_switch "pre") (eq .Cluster.ConfigItems.nlb_switch "exec") }}
           - "-forwarded-headers=X-Forwarded-For,X-Forwarded-Proto=https,X-Forwarded-Port=443"
           - "-forwarded-headers-exclude-cidrs=10.2.0.0/15,{{ .Values.vpc_ipv4_cidr}}"
 {{ end }}
-{{ if .ConfigItems.skipper_ingress_inline_routes }}
-          - "-inline-routes={{ .ConfigItems.skipper_ingress_inline_routes }}"
+{{ if .Cluster.ConfigItems.skipper_ingress_inline_routes }}
+          - "-inline-routes={{ .Cluster.ConfigItems.skipper_ingress_inline_routes }}"
 {{ end }}
-{{ if .ConfigItems.skipper_ingress_refuse_payload }}
-          {{ range $pattern := split .ConfigItems.skipper_ingress_refuse_payload "[cf724afc]" }}
+{{ if .Cluster.ConfigItems.skipper_ingress_refuse_payload }}
+          {{ range $pattern := split .Cluster.ConfigItems.skipper_ingress_refuse_payload "[cf724afc]" }}
           - "-refuse-payload={{ $pattern }}"
           {{ end }}
+{{ end }}
+{{ if index . "args" }}
+          {{ range $arg := split .args "[cf724afc]" }}
+          - "{{ $arg }}"
+          {{ end }}
 {{ end }}
         resources:
 {{ if and (eq .Cluster.ConfigItems.enable_dedicate_nodepool_skipper "true") (eq .Cluster.ConfigItems.skipper_ingress_binpack "true") }}
-{{ $cpu_requests := sumQuantities .Cluster.ConfigItems.skipper_ingress_cpu (printf "-%s"  .Cluster.ConfigItems.teapot_admission_controller_daemonset_reserved_cpu) (printf "-%s" .Cluster.ConfigItems.kubelet_system_reserved_cpu) (printf "-%s" .Cluster.ConfigItems.kubelet_kube_reserved_cpu) }}
+{{ $cpu_requests := sumQuantities .Cluster.ConfigItems.skipper_ingress_cpu (printf "-%s" .Cluster.ConfigItems.teapot_admission_controller_daemonset_reserved_cpu) (printf "-%s" .Cluster.ConfigItems.kubelet_system_reserved_cpu) (printf "-%s" .Cluster.ConfigItems.kubelet_kube_reserved_cpu) }}
```

Comment:
Oh I wasn't aware of that. So we can actually just set the configuration item to the resource of the node and this calculates the resources that can be used by skipper-ingress.

Reply:
This is only with `skipper_ingress_binpack=true` but this didn't work 100% so not enabled in most clusters yet. I was hoping to look into this with Karpenter as there seem to be a miscalculation with Cluster Autoscaler when I investigated it months ago.